docs/source_install.md

```diff
@@ -0,0 +1,122 @@
+# Build from Source
+
+- [Linux](#linux)
+- [OSX](#osx)
+
+## Linux
+
+Make sure you have [node.js](https://nodejs.org/en/) and [npm](https://www.npmjs.com/). If you don't, use your linux distro's package manager to get them.
+
+Get [zeromq](http://zeromq.org/) using the package manager of your distro. For example here is the package in [Archlinux](https://www.archlinux.org/packages/community/x86_64/zeromq/) and in [Ubuntu](https://packages.ubuntu.com/source/trusty/libs/zeromq).
+
+Also get [sqlcipher](https://www.zetetic.net/sqlcipher/):
+
+- If you're running Archlinux you can install the [package](https://www.archlinux.org/packages/community/x86_64/sqlcipher/) with `pacman`.
+
+- If you're running Ubuntu you will need to install [libsqlcipher-dev](https://packages.ubuntu.com/trusty/libdevel/libsqlcipher-dev) with `apt-get` instead.
+
+Install electron and any other npm dependencies by:
+
+```
+npm install --runtime=electron --target=1.8.4
+```
+
+Create a new [virtual environment](http://docs.python-guide.org/en/latest/dev/virtualenvs/) to install all the python dependencies. If you don't have `mkvirtualenv` then check how to get it depending on your distribution. [Here](http://exponential.io/blog/2015/02/10/install-virtualenv-and-virtualenvwrapper-on-ubuntu/) is a guide for Ubuntu and [here](https://wiki.archlinux.org/index.php/Python/Virtual_environment) is one for ArchLinux.
+
+```
+mkvirtualenv rotkehlchen
+```
+
+Then install all the python requirements by doing:
+
+```
+pip install -r requirements.txt
+```
+
+Now to start the application you need to type `npm start`.
+
+If you get runtime errors about a Node version mismatch, you can try to rebuild the electron
+modules like this:
+
+```
+./node_modules/.bin/electron-rebuild
```

Comment:
For some people they will need to put the `PYTHON=/path/to/python2.7` here.  Travis packaging fails without it. But let's keep it simple for now, and if someone complains I can add it as a conditional step.